research-software-directory/RSD-as-a-service
# 113 edit software contributors

# Edit software - contributors section

Closes #113 

Changes proposed in this pull request:

* User can find contributors registered in RSD or in ORCID database
* User can add new contributors using 'create' button that appears if contributor is not present in RSD or ORCID database

How to test:
* `docker-compose down --volumes`: remove old database
* `docker-compose build`: build all containers
* `docker-compose up`: run new instance
*  run data-migration: 
```bash
cd data-migration
docker-compose build
docker-compose up
```
* Start RSD on localhost and login using SURFconext test account proferssor1
* Create new software using + button at the top right
* Publish software using publish switch and save
* Navigate to contributors, you should see notification like this
![image](https://user-images.githubusercontent.com/9204081/154223485-5bbc7b41-d9a1-4722-a327-4135bdc47785.png)
* Add Jason Maassen as contributor, using its RSD record :-)
* Add Maaike de Jong as contributor, using her ORCID record
* Add Ewan Johanssson as contributor by creating new contributor (note! create button should appear). Try to upload an image.
* Edit one of contributors
* Delete one of contributors
* Preview changes using View button at the top of the page
* Check one of the existing software for editing. For this you will need to assign ownership manually in the database. Note the UUID used in the example below should be replaced with actual values from your local RSD database

```SQL
-- LOGIN FIRST THEN NOTE ID
select id from account;

-- THIS TABLE SHOULD BE EMPTY AT THE START
select * from maintainer_for_software;

-- SELECT SOFTWARE by slug to which you want to assign maintainer to 
select id from software 
	WHERE slug in ('dianna','ggir','xenon');

-- BECOME MAINTAINER, insert maintainer id you took from account table and software id's you took from software table
-- NOTE! the ids (account id and software id) are changing on each new creation of RSD database. During data migration new uuid to software are assigned
-- The account id is received from SURFconnext. It is not guaranteed to be "permanent" in test enviroment with test accounts 
INSERT into 
	maintainer_for_software (maintainer,software)
VALUES 
	('ba77fea5-9361-4978-af8e-1489ef163e22','10ae37db-83af-4f55-951f-7ce8b1a16e93'),
	('ba77fea5-9361-4978-af8e-1489ef163e22','9aa1e3b9-8c79-4c18-ae56-1350025414cc'),
	('ba77fea5-9361-4978-af8e-1489ef163e22','ac419ca9-8835-431f-9ba7-8c4ec85d09b5')
;

-- VALIDATE your insert worked properly, you should see new entries
select * from maintainer_for_software;
```

* For Xenon the contributors page should look similair to example below
![image](https://user-images.githubusercontent.com/9204081/154262229-28a2676c-b2ae-4e81-80f8-94ad7cfd736d.png)

PR Checklist:

*   [n/a] Update documentation
